src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -23,8 +24,8 @@ public UserController(UserService userService,
 
     @GetMapping("/by-email")
     public UserResponseDto findByEmail(@RequestParam String email) {
-        User user = userService.findByEmail(email).orElseThrow(
-                () -> new RuntimeException("Invalid email"));
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new DataProcessingException("Invalid email"));
```

Comment:
DataProcessingException means that we encountered problems with our database. Further such exception will produce response with 500 code which means it's our fault, user made proper request.
Here we should use other exception which will lead to 400 code to give user response that we're fine but his request is not valid
```suggestion
                .orElseThrow(() -> new NoSuchElementException("User with email " + email + " doesn't exist"));
```